src/main/java/com/thealgorithms/misc/FourSumProblem.java

```diff
@@ -0,0 +1,45 @@
+package com.thealgorithms.misc;
+import java.util.*;
+
+public class FourSumProblem {
+    public static List<List<Integer>> fourSum(int[] nums, int target) {
+        int n = nums.length; // size of the array
+        Set<List<Integer>> set = new HashSet<>();
+
+        // checking all possible quadruplets:
+        for (int i = 0; i < n; i++) {
```

Comment:
This approach does not have a good time complexity. Try to achieve better time complexity